tools/src/commands/PublishProjectTemplates.ts

```diff
@@ -126,31 +141,39 @@ async function action(options) {
 
     const moreArgs: string[] = [];
 
-    if (tag) {
-      // Assign custom tag in the publish command, so we don't accidentally publish as latest.
-      moreArgs.push('--tag', tag);
-    }
+    // Assign custom tag in the publish command, so we don't accidentally publish as latest.
+    moreArgs.push('--tag', npmPublishTag);
 
     // Publish to NPM registry
+    console.log('Executing command: npm publish ' + ['--access', 'public', ...moreArgs].join(' '));
     options.dry ||
       (await spawnAsync('npm', ['publish', '--access', 'public', ...moreArgs], {
         stdio: 'inherit',
         cwd: template.path,
       }));
 
-    if (tag && (await shouldAssignLatestTagAsync(template.name, newVersion))) {
+    if (tags.length > 1) {
```

Comment:
```suggestion
    if (tags.length === 2) {
```

slightly more clear that this is only intended to handle 2 tags and not n tags

Reply:
Fixed